src/index.ts

```diff
@@ -74,27 +74,27 @@ export const pathJoin = (dir: string, value: string): string => {
 
 export const exists = async (dir: string): Promise<boolean> => {
   if (urlParse(dir) !== undefined) {
-    try {
-      return await retry(
-        async (): Promise<boolean> => {
-          const { status } = await fetch(dir)
-
-          if (status !== 200) {
-            const msg = `StatusCode "${status}" for "${dir}"`
-
-            throw new Error(msg)
-          }
-
+    let retries = 0
+    const retryTimes = 3
+    while (retries++ < retryTimes) {
+      try {
+        const { status } = await fetch(dir, {
+          method: "HEAD",
+          timeout: 60 * 1000
+        })
+        if (status === 200) {
           return true
-        },
-        {
-          retries: 3,
-          factor: 1
+        } 
```

Comment:
For not 200 status trials, can we collect the result and report it after retry finished.